packages/main/src/requests/request.ts

```diff
@@ -76,6 +77,35 @@ export async function getHeaders(url: RequestUrl): Promise<Headers> {
   headers.append("Content-Type", "application/json");
   headers.append("x-goog-api-client", getClientHeaders(url.requestOptions));
   headers.append("x-goog-api-key", url.apiKey);
+
+  let customHeaders = url.requestOptions.customHeaders;
+  if (customHeaders) {
+    if (!(customHeaders instanceof Headers)) {
+      try {
+        customHeaders = new Headers(customHeaders);
+      } catch (e) {
+        throw new GoogleGenerativeAIRequestInputError(
+          `unable to convert customHeaders value ${JSON.stringify(
+            customHeaders,
+          )} to Headers: ${e.message}`,
+        );
+      }
+    }
+
+    for (const [headerName, headerValue] of customHeaders.entries()) {
+      if (
+        headerName === "x-goog-api-client" ||
+        headerName === "x-goog-api-key"
+      ) {
+        throw new GoogleGenerativeAIRequestInputError(
+          `cannot set reserved header name ${headerName}`,
```

Comment:
1. Capitalize the message unless it's going to be added to the end of another message string.
2. They can actually add to x-goog-api-client using the `apiClient` field, just not this one (it does it with spaces and no commas), so maybe that one needs a different message.